### ISSUE TITLE
Support multiple active tokens (TWO) for same set of user provided input

### ISSUE THREAD

**Issue by WickramBug** [wso2/api-manager#1819]
### Problem

In the current setup, the user can get a new token only after this Token1 is expired OR if the user changes the inputs and gives a different value for the "scope" parameter.
In this setup, users might face a scenario where they have an access token with a short lifetime left (like a few seconds) that can lead to API failure because, by the time the request is received on GW, the token might have already expired.

### Solution

Let's say the user created a token (Token1) that has a lifetime of 60 mins. 

The need is that if a user sends a token request after X time, say 30 mins or 45 mins (the time could be configurable) then the user will get a new token Token2 which will again be active for 60 mins and Token1 should also be active for the remaining 30 mins or 15 mins respectively of its lifespan.

That way, users can issue token requests at some known intervals and have the confidence that they will get new active tokens.

In a nutshell,
1. Token expiry time should be 15 mins
2. New tokens should be generated at least 5 mins before the other key expiration.
3. Active tokens should not be more than 2. At any time during the lifespan of a token(15 mins), there should be only two active tokens.

### Affected Component

APIM

### Version

3.2.0

**Comment by dulithsenanayake** [comment]
Hi all,

Closing this as the fix is released.

Thanks!
Dulith
